tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1623,30 +1238,64 @@ void ROOT::Experimental::Internal::RNTupleFileWriter::WriteBareFileSkeleton(int
 
 void ROOT::Experimental::Internal::RNTupleFileWriter::WriteTFileStreamerInfo()
 {
+   // The streamer info record is a TList of TStreamerInfo object.  We cannot use
+   // RNTupleSerializer::SerializeStreamerInfos because that uses TBufferIO::WriteObject.
+   // This would prepend the streamed TList with self-decription information.
+   // The streamer info record is just the streamed TList.
+
+   // For classes with read rules, we additionally need to store the read rules.
```

Comment:
Currently, `TFile` does not store the read rule for the following reasons:
* They are 'relatively' large (compare to the size of the `TStramerInfo`)
* Reading them is slow (they are strings).
* We have not implemented the code to turn those read rules into actual action at run-time (only at dictionary generation time).
* Related to the previous point, the code that generate assumes that the class they are applied to has a header file/declaration. (I.e. in which case, the read rule action/function has been generated as part of the dictionary.
* A lot of the rules apply to the transient members (which would likely only exist in the case where we have the code for the class and hence the dictionary).
* etc.

Reply:
That makes sense. I overlooked the corresponding code in TFile is inactive.